contents/swift/welcome-to-swift.md

```diff
@@ -144,9 +147,45 @@ Here are some of the benefits of using `guard` statement over `if-let` statement
 1. Unlike the `if-let` statement, using the `guard` statement causes `past` to remain defined and can be used till the function exits.
 1. While using `if-let` statements can lead to deeply nested `if-let` statements (i.e. pyramid of doom), `guard` statements allow us to have the happy path to be not indented, thereby increasing code readability.
 
-## Data Types
+### Benefit: Handle Post-operations easily
+
+The `defer` key word in Swift provides an easy and safe way to execute some code before leaving current scope. It is helpful when you need to do post-operations in a function which has many points of return.
+
+The following code is an example of using a file:
+```swift
+let fileDescriptor = open(url.path, O_EVTONLY)
+if fileDescriptor == -1 {
+	close(fileDescriptor)
+	return "Failed"
+}
+
+// Use file descriptor
+
+close(fileDescriptor)
+return "Success"
+```
+
+As you can see from above, we have to close the `fileDescriptor` for every case we consider. This can be problematic when the number of cases increases. Instead, we can use the `defer` statement:
```

Comment:
number of cases increase

Reply:
I think it should be "increase" because the subject is "number", and is not plural

Reply:
Yup my bad! You are right.. (sounded off when I said it out in my mind)